enzyme/Enzyme/EnzymeLogic.cpp

```diff
@@ -2780,9 +2780,17 @@ Function *EnzymeLogic::CreatePrimalAndGradient(
 
     if (mode == DerivativeMode::ReverseModeCombined) {
 
-      FunctionType *FTy =
-          FunctionType::get(StructType::get(todiff->getContext(), {res.second}),
-                            res.first, todiff->getFunctionType()->isVarArg());
+      FunctionType *FTy;
+      if (!res.second.empty()) {
+        FTy = FunctionType::get(
+            StructType::get(todiff->getContext(), {res.second}), res.first,
```

Comment:
I'd just have the conditional set the function return type, and move the common code outside the conditional.